src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/openapi/TokenCertificatesApiControllerTest.java

```diff
@@ -0,0 +1,242 @@
+import org.niis.xroad.restapi.service.KeyNotFoundException;
+import org.niis.xroad.restapi.service.TokenCertificateService;
+import org.niis.xroad.restapi.util.CertificateTestUtils;
+import org.niis.xroad.restapi.util.TestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.security.cert.X509Certificate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static ee.ria.xroad.common.ErrorCodes.SIGNER_X;
+import static ee.ria.xroad.common.ErrorCodes.X_CERT_EXISTS;
+import static ee.ria.xroad.common.ErrorCodes.X_CSR_NOT_FOUND;
+import static ee.ria.xroad.common.ErrorCodes.X_INCORRECT_CERTIFICATE;
+import static ee.ria.xroad.common.ErrorCodes.X_KEY_NOT_FOUND;
+import static ee.ria.xroad.common.ErrorCodes.X_WRONG_CERT_USAGE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+
+/**
+ * test certificates api
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureTestDatabase
+@Transactional
+@Slf4j
+public class TokenCertificatesApiControllerTest {
+
+    @MockBean
+    private SignerProxyFacade signerProxyFacade;
+
+    @MockBean
+    private GlobalConfFacade globalConfFacade;
+
+    @Autowired
+    private TokenCertificatesApiController tokenCertificatesApiController;
+
+    @Before
+    public void setup() throws Exception {
+        doAnswer(answer -> "key-id").when(signerProxyFacade).importCert(any(), any(), any());
+        doAnswer(answer -> null).when(globalConfFacade).verifyValidity();
+        doAnswer(answer -> TestUtils.INSTANCE_FI).when(globalConfFacade).getInstanceIdentifier();
+        doAnswer(answer -> TestUtils.getM1Ss1ClientId()).when(globalConfFacade).getSubjectName(any(), any());
+    }
+
+    @Test
+    @WithMockUser(authorities = "IMPORT_SIGN_CERT")
+    public void addSignCertificate() {
+        Resource body = CertificateTestUtils.getResource(CertificateTestUtils.getMockCertificateBytes());
+        ResponseEntity<CertificateDetails> response = tokenCertificatesApiController.addCertificate(body);
+        CertificateDetails addedCert = response.getBody();
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertCertificateDetails(addedCert);
+    }
+
+    @Test
```

Comment:
Would probably be good to add a test for authentication & certificate type (attempt to import auth cert, only rights to import sign cert -> AccessDenied)

Reply:
Added tests [here](https://github.com/nordic-institute/X-Road/pull/268/commits/e241e058684f68a6be0def5f7fd946942748773f#diff-256e66ddc9c48e2da44062140bb3047bR115) and [here](https://github.com/nordic-institute/X-Road/pull/268/commits/e241e058684f68a6be0def5f7fd946942748773f#diff-256e66ddc9c48e2da44062140bb3047bR134)